### PR TITLE
PollingController: add option to poll by blockTracker

### PR DESCRIPTION
**Pull request by adonesky1**
Addresses: https://github.com/MetaMask/MetaMask-planning/issues/1788

[@matthewwalsh0](https://github.com/matthewwalsh0) and the Confirmation Systems Team is hoping to use the `PollingController` for a new `UserOperationController` but are needing to tie polling intervals to block times rather than a static interval length. This is achievable using the vanilla blockTracker and adding callbacks to its listeners but we will want to multiplex it for multichain and it so it makes sense to do so using the `PollingController` abstraction since they are basically doing the same thing but just using a different source for the interval of execution.

### CODE REVIEW

**Comment by jiexi** [comment]
It's almost like we need another abstraction that handles just the idea of polling tokens, starting, and stopping. And two implementations of that for Polling by setTimeout and polling by subscription with listener

**Comment by adonesky1** [comment]
> It's almost like we need another abstraction that handles just the idea of polling tokens, starting, and stopping. And two implementations of that for Polling by setTimeout and polling by subscription with listener

I agree this would be a more elegant solution. Not sure if we need this to be flexible to more types of polling. All a question of time and effort imo.

File: packages/polling-controller/src/PollingController.ts [hunk 1]
```diff
@@ -38,24 +41,53 @@ function PollingControllerMixin<TBase extends Constructor>(Base: TBase) {
...
      */
     setIntervalLength(length: number) {
       this.#intervalLength = length;
+
+      // setting and using an interval is mutually exclusive with polling on new blocks
+      this.#getNetworkClientById = undefined;
+    }
+
+    setPollWithBlockTracker(
+      getNetworkClientById: (networkClientId: NetworkClientId) => NetworkClient,
```
Comment (matthewwalsh0) [inline packages/polling-controller/src/PollingController.ts:81]: I wasn't sure if this argument would be best as a `ControllerMessenger` as this `(Network)PollingController` can know how to get a network client given an ID, plus it encourages messenger adoption.

But I suppose the counter-argument is that it requires all child controllers to `BaseController2` or have a messenger.

**Comment by adonesky1** [comment]
Superceded by https://github.com/MetaMask/core/pull/3636

**Pull request closed without merging**